tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/traversal/Filter.java

```diff
@@ -0,0 +1,285 @@
+     * patterns for assets.
+     *
+     * @param path The asset path to check.
+     * @return {@code true} if the path should be included, {@code false} otherwise.
+     */
+    private boolean useAsset(String path) {
+
+        FileSystem fileSystem = FileSystems.getDefault();
+
+        // Check if the path should be used according to the excludes
+        for (PathMatcher exclude : assetExcludes) {
+            if (exclude.matches(fileSystem.getPath(path))) {
+                return false;
+            }
+        }
+
+        // Check if the path should be used according to the includes
+        if (assetIncludes.isEmpty()) {
+            return true;
+        }
+
+        for (PathMatcher include : assetIncludes) {
+            if (include.matches(fileSystem.getPath(path))) {
+                return true;
+            }
+        }
+
+        // If no include patterns matched, exclude the path by default
+        return false;
+    }
+
+    /**
+     * Creates a new builder for constructing a {@code Filter} instance.
+     *
+     * @return A new builder instance.
+     */
+    public static Filter.Builder builder() {
+        return new Filter.Builder();
+    }
+
+    /**
+     * A builder for constructing a {@code Filter} instance with includes and excludes patterns for
+     * folders and assets.
+     */
+    public static class Builder {
+
+        private final Set<PathMatcher> includeFolderPatterns = new HashSet<>();
+        private final Set<PathMatcher> includeAssetPatterns = new HashSet<>();
+        private final Set<PathMatcher> excludeFolderPatterns = new HashSet<>();
+        private final Set<PathMatcher> excludeAssetPatterns = new HashSet<>();
+
+        private String rootPath = "/";
+
+        private Builder() {
+        }
+
+        /**
+         * Sets the root path for the filter. The root path is used to remove the root path prefix
+         * from the folder and asset paths before applying the filter.
+         *
+         * @param rootPath The root path to set.
+         * @return The builder instance.
+         */
+        public Builder rootPath(String rootPath) {
+            this.rootPath = rootPath;
+            return this;
+        }
+
+        /**
+         * Adds include patterns for folders.
+         *
+         * @param include The include patterns for folders.
+         * @return The builder instance.
+         */
+        public Builder includeFolder(String... include) {
+            for (String pattern : include) {
+                includeFolderPatterns.add(
+                        FileSystems.getDefault().getPathMatcher("glob:" + pattern)
+                );
+            }
+            return this;
+        }
+
+        /**
+         * Adds include patterns for assets.
+         *
+         * @param include The include patterns for assets.
+         * @return The builder instance.
+         */
+        public Builder includeAsset(String... include) {
```

Comment:
I think you can do, Stream.of(include).forEach(pattern -> ncludeAssetPatterns.add(
                        FileSystems.getDefault().getPathMatcher("glob:" + pattern))

Reply:
Done